app/jobs/notifications/remove_each_job.rb

```diff
@@ -0,0 +1,16 @@
+module Notifications
+  class RemoveEachJob < ApplicationJob
+    queue_as :remove_each_notifications
+
+    def perform(notifiable_collection_ids, service = Notifications::RemoveEach)
+      notifications = Notification.where(notifiable_id: notifiable_collection_ids)
+      return unless notifications.any?
+
+      service.call(self.class.notifiable_collection(notifications))
+    end
+
+    def self.notifiable_collection(notifications)
```

Comment:
I think this should be part of the service internal logic, not the job. This is because you're hiding the fact that the service now wants a hash structured in a certain way instead of a regular ActiveRecord collection. Let the service receive the collection and then internally you extract what you need :-)

Reply:
It's also not safe to query only by `notifiable_id`. There could be notifications with the same `notifiable_id` but different `notifiable_type`. So you either have to pass `notifiable_type`, or just change the query to the `Notification.where(notifiable_id: notifiable_collection_ids, notifiable_type: "Mention")`. I think that the second option is enough for now.

Reply:
Yes, it was over complicated. I have simplified, I needed just one request. Though, as there are some risks with seralization, I had to pass ids to the job and not AR objects.

Reply:
Yep, definitely `ids` to the job, it's one of the points of the refactoring :eyes: